plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/inner-blocks/clear-button/frontend.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { store, getContext } from '@wordpress/interactivity';
+
+/**
+ * Internal dependencies
+ */
+import { ProductFiltersContext } from '../../frontend';
+
+type ClearButtonContext = {
+	parent: string;
+};
+
+const { actions } = store( 'woocommerce/product-filters', {
```

Comment:
You can import the main type of the store here, to type `actions.navigate`.


```suggestion
const { actions } = store< ProductFilter >( 'woocommerce/product-filters', {
```

Also, you can join it to its own inferred types, or other store part types, depending on what this store part should be aware of.

More information on how to work with TypeScript in the Interactivity API in [this guide](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/core-concepts/using-typescript/).

Reply:
Addressed in d9424fc5061c

Reply:
Thanks.

What are your thoughts about this way of working with types and sharing them among the different store parts?